src/UiDatepicker.vue

```diff
@@ -66,12 +66,14 @@
                     v-show="!disabled"
                 >
                     <ui-datepicker-calendar
+                        ref="datepickerCalendar"
```

Comment:
Maybe we don't need the `ref` + `reinitialize()` if we just treat `defaultView` as a default only (e.g. it starts with the year for the first time only, and after the user selects a year and date the view will be date, until they click the year selector).

Thoughts? 🤔

Reply:
Unfortunately we still do (at least in my use case).
From what I understood, the DatepickerCalendar is appended on input created. Without this, the datepicker shows with year selector only the first time. Every following time it opens where it was left (so, `showYearPicker: false`).
So effectively user is shown the year picker only once, and then has to initialize it himself.
I'm afraid my users are not smart enough to guess that clicking on a year will lead you to year selection list.
Lemme know what you think, and I'll try to resolve things tomorrow at work. Cheers!

Reply:
I see. How about this:

On `UiDatepickerCalendar`

- Rename `defaultView` prop to `currentView`
- Move `showYearPicker` from data to a computed value based on `currentView`, e.g. 
```js
showYearPicker() {
	return this.currentView === 'year';
}
```
- Add `showDatePicker` computed value, same as above but checking for `'date'` in the current view
- In places where we assign `showYearPicker = false`, emit an update event, e.g.
```js
$emit('update:currentView', 'date')
```
- In places where we assign `showYearPicker = true`, emit another update event as above, but setting `currentView` to `'year'`
- Update the `showYearPicker` watcher to watch `currentView` instead, and check if it is `'year'` to scroll to the selected year

By using these 'update' events, we hand over control of the `currentView` prop to the parent, who can change it directly, without going through a method, e.g. `reinitialize`.

So, on `UiDatepicker`, we have,

- A new prop `defaultView`, and a new data value `calendarView`, which is initially set to the `defaultView` prop.
- Pass the `calendarView` data value to the calendar component with the [`.sync` modifier](https://vuejs.org/v2/guide/components-custom-events.html#sync-Modifier) so it's automatically synced when the component emits 'update' events:

```html
<ui-datepicker-calendar
  	// other props
	:current-view.sync="calendarView"
></ui-datepicker-calendar>
```

We can then reset the `calendarView` data value when the datepicker is closed, so that the next time it's opened it shows what we specify in `defaultView`:

In the [`closePicker()`](https://github.com/entioentio/Keen-UI/blob/next/src/UiDatepicker.vue#L315) method:

```js
this.getPicker().close();
this.calendarView = this.defaultView
```

Let me know if that makes sense.

Reply:
Perfect sense. I kind of wanted to hack it around without changing your code too far. Solution you proposed makes it more robust. Nice.

Reply:
Thanks for suggestions, @JosephusPaye It's implemented and it works. Btw. I had no idea about `.sync` modifier. How handy!